Update Docs for new release

carltongibson opened the pull request:
For #595. Preparing the docs. 

To Do:

- [x] changelog.txt
- [x] contributing.txt

Others seems more or less okay.

diox commented:
I think it's good to have a high level overview in `changelog.txt`. Users interested in the details can find the github comparison link by themselves, but it's useful to have a quick overview of the changes directly in the docs for people that simply want to see if they should apply the upgrade or not, for instance.

carltongibson commented:
Totally agree. It's just a question of what's "Quick Overview" and what's not. The compatibility with 1.7 & 1.8 seem like headlines — not sure how much else to put. 

Aside: is there an easy way to see issues/pulls closed since tag? (This would help.)

diox commented:
There are various tools for this - search "git changelog generator" or "github changelog generator", something along those lines. Don't include the full commits, just a one line explaining each PR that we merged should be enough (I don't expect we merged many)

carltongibson commented:
Nice. This one looks like the bad-boy https://github.com/skywinder/github-changelog-generator — I'll see what it comes up with.

carltongibson commented:
Okay — there's a list of the merged PRs, very roughly `sed`-ed into rST format. 

(There's a few escapes from the markdown that may not be right, but I can tidy those if the general approach is ok)

diox commented on docs/changelog.txt:
```diff
@@ -1,6 +1,82 @@
+
+`Full Changelog <https://github.com/django-compressor/django-compressor/compare/1.4...HEAD>`_
+
+- Fix typo and restyle manage command examples `\#604 <https://github.com/django-compressor/django-compressor/pull/604>`_ (`alimony <https://github.com/alimony>`_)
```
We can ignore this one

diox commented:
It looks good, but a few of those I don't think we need, and although having the name of the author is good, I don't think we need the links to the PR or to the author github profile.

diox commented on docs/changelog.txt:
```diff
@@ -1,6 +1,82 @@
+
+- Fix typo and restyle manage command examples `\#604 <https://github.com/django-compressor/django-compressor/pull/604>`_ (`alimony <https://github.com/alimony>`_)
+
+- Fix compress command for 1.8 `\#601 <https://github.com/django-compressor/django-compressor/pull/601>`_ (`carltongibson <https://github.com/carltongibson>`_)
```
Add "Django" before "1.8".

diox commented on docs/changelog.txt:
```diff
@@ -1,6 +1,82 @@
+
+- Fix compress command for 1.8 `\#601 <https://github.com/django-compressor/django-compressor/pull/601>`_ (`carltongibson <https://github.com/carltongibson>`_)
+
+- Fix build `\#599 <https://github.com/django-compressor/django-compressor/pull/599>`_ (`carltongibson <https://github.com/carltongibson>`_)
```
We can ignore this one

diox commented on docs/changelog.txt:
```diff
@@ -1,6 +1,82 @@
+
+- Fix build `\#599 <https://github.com/django-compressor/django-compressor/pull/599>`_ (`carltongibson <https://github.com/carltongibson>`_)
+
+- Update settings.txt `\#584 <https://github.com/django-compressor/django-compressor/pull/584>`_ (`johnfraney <https://github.com/johnfraney>`_)
```
We can ignore this one

diox commented on docs/changelog.txt:
```diff
@@ -1,6 +1,82 @@
+
+- Fix reading UTF-8 files which have BOM `\#568 <https://github.com/django-compressor/django-compressor/pull/568>`_ (`JereMalinen <https://github.com/JereMalinen>`_)
+
+- shlex.quote/pipes.quote is not compatible with Windows `\#561 <https://github.com/django-compressor/django-compressor/pull/561>`_ (`thenewguy <https://github.com/thenewguy>`_)
```
Rephrase with "Fix incompatibility with Windows (shell_quote is not supported)" or something similar

diox commented on docs/changelog.txt:
```diff
@@ -1,6 +1,82 @@
+
+- Run automated tests on Django 1.7 `\#556 <https://github.com/django-compressor/django-compressor/pull/556>`_ (`frewsxcv <https://github.com/frewsxcv>`_)
+
+- Patch 1 `\#555 <https://github.com/django-compressor/django-compressor/pull/555>`_ (`francescozaia <https://github.com/francescozaia>`_)
```
We can ignore this one

diox commented on docs/changelog.txt:
```diff
@@ -1,6 +1,82 @@
+
+- Patch 1 `\#555 <https://github.com/django-compressor/django-compressor/pull/555>`_ (`francescozaia <https://github.com/francescozaia>`_)
+
+- Fix AttributeError in offline compression `\#554 <https://github.com/django-compressor/django-compressor/pull/554>`_ (`cbjadwani <https://github.com/cbjadwani>`_)
```
Change text to "Ignore non-existent `{{ block.super }}` in offline compression instead of raising `AttributeError`"

diox commented on docs/changelog.txt:
```diff
@@ -1,6 +1,82 @@
+
+- Support for clean-css `\#551 <https://github.com/django-compressor/django-compressor/pull/551>`_ (`vovanbo <https://github.com/vovanbo>`_)
+
+- Types at readme \(-engine should be --engine\) `\#540 <https://github.com/django-compressor/django-compressor/pull/540>`_ (`lukaszb <https://github.com/lukaszb>`_)
```
We can ignore this one

diox commented on docs/changelog.txt:
```diff
@@ -1,6 +1,82 @@
+
+- Types at readme \(-engine should be --engine\) `\#540 <https://github.com/django-compressor/django-compressor/pull/540>`_ (`lukaszb <https://github.com/lukaszb>`_)
+
+- Fix typo `\#539 <https://github.com/django-compressor/django-compressor/pull/539>`_ (`alimony <https://github.com/alimony>`_)
```
We can ignore this one

diox commented on docs/changelog.txt:
```diff
@@ -1,6 +1,82 @@
+
+- Fix typo `\#539 <https://github.com/django-compressor/django-compressor/pull/539>`_ (`alimony <https://github.com/alimony>`_)
+
+- Fix link markup `\#530 <https://github.com/django-compressor/django-compressor/pull/530>`_ (`merwok <https://github.com/merwok>`_)
```
We can ignore this one

diox commented on docs/changelog.txt:
```diff
@@ -1,6 +1,82 @@
+
+- Use django.template.loader.get\_template\(\) in compress management command `\#282 <https://github.com/django-compressor/django-compressor/pull/282>`_ (`mattpatey <https://github.com/mattpatey>`_)
+
+- Compress selected templates `\#248 <https://github.com/django-compressor/django-compressor/pull/248>`_ (`jaap3 <https://github.com/jaap3>`_)
```
This PR is closed, not merged. It should not be included.

diox commented on docs/changelog.txt:
```diff
@@ -1,6 +1,82 @@
+
+- Compress selected templates `\#248 <https://github.com/django-compressor/django-compressor/pull/248>`_ (`jaap3 <https://github.com/jaap3>`_)
+
+- re-implementing default documented behaviour for uncompressable files, fixes \#241 `\#246 <https://github.com/django-compressor/django-compressor/pull/246>`_ (`ssaboum <https://github.com/ssaboum>`_)
```
This PR is closed, not merged. It should not be included.

diox commented on docs/changelog.txt:
```diff
@@ -1,6 +1,82 @@
+
+- Don't leave `compressed` in the render context `\#354 <https://github.com/django-compressor/django-compressor/pull/354>`_ (`gavinwahl <https://github.com/gavinwahl>`_)
+
+- Adding support to ignore some file paths during offline compression `\#335 <https://github.com/django-compressor/django-compressor/pull/335>`_ (`ayadav <https://github.com/ayadav>`_)
```
This PR is closed, not merged. It should not be included.

diox commented on docs/changelog.txt:
```diff
@@ -1,6 +1,82 @@
+
+- Remove compatibility with old 'staticfiles' app `\#525 <https://github.com/django-compressor/django-compressor/pull/525>`_ (`diox <https://github.com/diox>`_)
+
+- Use get\_template\(\) when dealing with django templates `\#524 <https://github.com/django-compressor/django-compressor/pull/524>`_ (`diox <https://github.com/diox>`_)
```
Replace text with "In compress command, use get_template() instead of opening template files manually, fixing compatibility issues with custom template loaders".

diox commented on docs/changelog.txt:
```diff
@@ -1,6 +1,82 @@
+
+- Use get\_template\(\) when dealing with django templates `\#524 <https://github.com/django-compressor/django-compressor/pull/524>`_ (`diox <https://github.com/diox>`_)
+
+- Fix FilterBase clears self.type for subclasses `\#479 <https://github.com/django-compressor/django-compressor/pull/479>`_ (`iknite <https://github.com/iknite>`_)
```
Replace text with "Fix FilterBase so that does not override self.type for subclasses if filter_type is not specified at init" or something similar.

diox commented on docs/changelog.txt:
```diff
@@ -1,6 +1,82 @@
+
+- Remove unnecessary filename and existence checks in CssAbsoluteFilter `\#471 <https://github.com/django-compressor/django-compressor/pull/471>`_ (`agriffis <https://github.com/agriffis>`_)
+
+- Fix minor typographical error in docs/index `\#594 <https://github.com/django-compressor/django-compressor/pull/594>`_ (`varunsharma <https://github.com/varunsharma>`_)
```
We can ignore this one

diox commented on docs/changelog.txt:
```diff
@@ -1,6 +1,82 @@
+
+- Fix minor typographical error in docs/index `\#594 <https://github.com/django-compressor/django-compressor/pull/594>`_ (`varunsharma <https://github.com/varunsharma>`_)
+
+- Fix deprecation warning `\#548 <https://github.com/django-compressor/django-compressor/pull/548>`_ (`davidfischer-ch <https://github.com/davidfischer-ch>`_)
```
We can ignore this one

diox commented on docs/changelog.txt:
```diff
@@ -1,6 +1,82 @@
+
+- Fix deprecation warning `\#548 <https://github.com/django-compressor/django-compressor/pull/548>`_ (`davidfischer-ch <https://github.com/davidfischer-ch>`_)
+
+- Add "enabled" override to compress template tag `\#547 <https://github.com/django-compressor/django-compressor/pull/547>`_ (`davidfischer-ch <https://github.com/davidfischer-ch>`_)
```
This PR is closed, not merged. It should not be included.

carltongibson commented:
:-) Wow. That's a lot of comments. 

It looks like we're doing it by hand and deciding what's important and what's not is the way to go. I think this either needs to be done (religiously) as we go or else we give a  "Here are the highlights" and link  to the full GitHub change log. 

I'll adjust these.

diox commented:
I think by removing the typo fixes and small doc changes we'll end up with a good "here are the highlights" changelog :)

diox commented on docs/changelog.txt:
```diff
@@ -1,6 +1,82 @@
+
+- quote filename e.g. in case it contains spaces `\#535 <https://github.com/django-compressor/django-compressor/pull/535>`_ (`MikeAmy <https://github.com/MikeAmy>`_)
+
+- Put charset back into kwargs `\#523 <https://github.com/django-compressor/django-compressor/pull/523>`_ (`zmetcalf <https://github.com/zmetcalf>`_)
```
This PR is closed, not merged. It should not be included.

diox commented on docs/changelog.txt:
```diff
@@ -1,6 +1,82 @@
+
+- Add "enabled" override to compress template tag `\#547 <https://github.com/django-compressor/django-compressor/pull/547>`_ (`davidfischer-ch <https://github.com/davidfischer-ch>`_)
+
+- CSSO support `\#538 <https://github.com/django-compressor/django-compressor/pull/538>`_ (`vovanbo <https://github.com/vovanbo>`_)
```
This PR is closed, not merged. It should not be included.

diox commented on docs/changelog.txt:
```diff
@@ -1,6 +1,82 @@
+
+- CSSO support `\#538 <https://github.com/django-compressor/django-compressor/pull/538>`_ (`vovanbo <https://github.com/vovanbo>`_)
+
+- quote filename in command  `\#536 <https://github.com/django-compressor/django-compressor/pull/536>`_ (`MikeAmy <https://github.com/MikeAmy>`_)
```
This PR is closed, not merged. It should not be included.

diox commented on docs/changelog.txt:
```diff
@@ -1,6 +1,82 @@
+
+- quote filename in command  `\#536 <https://github.com/django-compressor/django-compressor/pull/536>`_ (`MikeAmy <https://github.com/MikeAmy>`_)
+
+- quote filename e.g. in case it contains spaces `\#535 <https://github.com/django-compressor/django-compressor/pull/535>`_ (`MikeAmy <https://github.com/MikeAmy>`_)
```
This PR is closed, not merged. It should not be included.

diox commented on docs/changelog.txt:
```diff
@@ -1,6 +1,82 @@
+
+- Put charset back into kwargs `\#523 <https://github.com/django-compressor/django-compressor/pull/523>`_ (`zmetcalf <https://github.com/zmetcalf>`_)
+
+- Fix bug with css compress for static files `\#504 <https://github.com/django-compressor/django-compressor/pull/504>`_ (`rogaha <https://github.com/rogaha>`_)
```
This PR is closed, not merged. It should not be included.

diox commented on docs/changelog.txt:
```diff
@@ -1,6 +1,82 @@
+
+- Fix bug with css compress for static files `\#504 <https://github.com/django-compressor/django-compressor/pull/504>`_ (`rogaha <https://github.com/rogaha>`_)
+
+- Support for async/defer tags `\#482 <https://github.com/django-compressor/django-compressor/pull/482>`_ (`hkpeprah <https://github.com/hkpeprah>`_)
```
This PR is closed, not merged. It should not be included.

diox commented on docs/changelog.txt:
```diff
@@ -1,6 +1,82 @@
+
+- Support for async/defer tags `\#482 <https://github.com/django-compressor/django-compressor/pull/482>`_ (`hkpeprah <https://github.com/hkpeprah>`_)
+
+- resolve \[issue 171\] : offline compressor to compress 'compress tags' if those are not in child template but in parent template. `\#436 <https://github.com/django-compressor/django-compressor/pull/436>`_ (`sjiitr <https://github.com/sjiitr>`_)
```
This PR is closed, not merged. It should not be included.

diox commented on docs/changelog.txt:
```diff
@@ -1,6 +1,82 @@
+
+- resolve \[issue 171\] : offline compressor to compress 'compress tags' if those are not in child template but in parent template. `\#436 <https://github.com/django-compressor/django-compressor/pull/436>`_ (`sjiitr <https://github.com/sjiitr>`_)
+
+- Don't leave `compressed` in the render context `\#354 <https://github.com/django-compressor/django-compressor/pull/354>`_ (`gavinwahl <https://github.com/gavinwahl>`_)
```
This PR is closed, not merged. It should not be included.

diox commented on docs/changelog.txt:
```diff
@@ -1,6 +1,82 @@
+
+- Adding support to ignore some file paths during offline compression `\#335 <https://github.com/django-compressor/django-compressor/pull/335>`_ (`ayadav <https://github.com/ayadav>`_)
+
+- Use django.template.loader.get\_template\(\) in compress management command `\#282 <https://github.com/django-compressor/django-compressor/pull/282>`_ (`mattpatey <https://github.com/mattpatey>`_)
```
This PR is closed, not merged. It should not be included.

diox commented:
Ok, this is a very good basis, thanks for doing this. I went over each line and added comments. In particular, be sure to remove the closed PR that were never merged. Let me know when you are done, thanks!

carltongibson commented:
It’s ok for a first pass. I think with use of labels and milestones I could put a filter together than would include the exact PRs wanted. (It would then be pretty easy to drawn up the list.) 




Anyhooo… ;-)

carltongibson commented:
@diox I updated the Changelog as per the comments. 

* I removed the links to the issues
* I left the links to the profiles in with the names — am happy to remove those or the names entirely, as you think.

I'll tick off the `changes` pending any comment you have. I'll look at `contributing` over the weekend.

carltongibson commented:
Hey @diox — Sorry for the pause here: I got ambushed by Easter without realising it was coming. 

I see the build failed on the last commit — but that was a few github timeouts — so I assume it would pass if restarted. 

More to the point though... 

I finally took the few minutes to look at the `contributing.txt`. Beyond removing the advice to follow jezdez — which whilst good is no longer relevant there — I'd consider moving people away from IRC: there's approximately nobody on the django-compressor channel at any one time — and not enough active contributors to make it good. 

Instead I'd advise opening an issue here for development issues and heading to StackOverflow for usage questions.

Before I make that change though is it something you'd agree with?

diox commented:
Sure, go ahead. Maybe keep the IRC channel but downplay it for the reasons you mention - I think it's still useful for potential contributors (I answer there if there are questions about contributing to compressor)

diox commented:
Also, I took a second look at the changelog and I believe we should remove the names+links entirely. But make sure the AUTHORS file is up to date.

carltongibson commented:
Hi @diox — Sorry for the long silence. I got totally nuked with work. 

I've adjusted the Change log and made a minimal change to the Contributing note. Hopefully that's okay for right now and we can push on with the release (which with 1.8 out I feel needs to go.)

diox commented:
Great, thanks!
